xen/arch/arm/coproc/coproc.c

```diff
@@ -341,6 +342,27 @@ int vcoproc_handle_node(struct domain *d, void *fdt,
     return res;
 }
 
+static int vcoproc_browse_node(struct domain *d,
+                               const struct dt_device_node *node)
+{
+    struct dt_device_node *child;
+    int res = 0;
```

Comment:
because on recursive call failure, you should not continue the loop, but report an error to a caller